src/Microsoft.OpenApi.Hidi/OpenApiService.cs

```diff
@@ -327,37 +328,51 @@ private static async Task<Stream> GetStream(string input, ILogger logger)
             return requestUrls;
         }
 
-        internal static async Task ValidateOpenApiDocument(string openapi, LogLevel loglevel)
+        internal static async Task ValidateOpenApiDocument(string openapi, LogLevel loglevel, CancellationToken cancellationToken)
         {
-            if (string.IsNullOrEmpty(openapi))
-            {
-                throw new ArgumentNullException(nameof(openapi));
-            }
             var logger = ConfigureLoggerInstance(loglevel);
-            var stream = await GetStream(openapi, logger);
 
-            OpenApiDocument document;
-            logger.LogTrace("Parsing the OpenApi file");
-            document = new OpenApiStreamReader(new OpenApiReaderSettings
+            try
             {
-                RuleSet = ValidationRuleSet.GetDefaultRuleSet()
-            }
-            ).Read(stream, out var context);
+                if (string.IsNullOrEmpty(openapi))
+                {
+                    throw new ArgumentNullException(nameof(openapi));
+                }
+                var stream = await GetStream(openapi, logger, cancellationToken);
 
-            if (context.Errors.Count != 0)
-            {
-                foreach (var error in context.Errors)
+                OpenApiDocument document;
+                logger.LogTrace("Parsing the OpenApi file");
+                document = new OpenApiStreamReader(new OpenApiReaderSettings
                 {
-                    Console.WriteLine(error.ToString());
+                    RuleSet = ValidationRuleSet.GetDefaultRuleSet()
+                }
+                ).Read(stream, out var context);
+
+                if (context.Errors.Count != 0)
+                {
+                    foreach (var error in context.Errors)
+                    {
+                        logger.LogError("OpenApi Parsing error: {message}", error.ToString());
+                        Console.WriteLine(error.ToString());
+                    }
                 }
-            }
 
-            var statsVisitor = new StatsVisitor();
-            var walker = new OpenApiWalker(statsVisitor);
-            walker.Walk(document);
+                var statsVisitor = new StatsVisitor();
+                var walker = new OpenApiWalker(statsVisitor);
+                walker.Walk(document);
+
+                logger.LogTrace("Finished walking through the OpenApi document. Generating a statistics report..");
+                Console.WriteLine(statsVisitor.GetStatisticsReport());
+            }
+            catch(Exception ex)
+            {
+#if DEBUG
+                logger.LogCritical(ex, ex.Message);
+#else
+                logger.LogCritical(ex.Message);
+#endif
+            }
```

Comment:
```suggestion
               return 1;
            }
```

Reply:
Why `return 1` instead of `return`?

Reply:
so the caller knows the process terminated with an error
exit code: 0 finished and successfull, anything else is error.
this is how a script, CI (ADO, github actions...) knows the process ran into an error.
https://github.com/dotnet/command-line-api/blob/main/docs/Process-termination-handling.md